tests/NCronJob.Tests/RunAtStartupJobTests.cs

```diff
@@ -0,0 +1,102 @@
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Shouldly;
+
+namespace NCronJob.Tests;
+
+public class RunAtStartupJobTests : JobIntegrationBase
+{
```

Comment:
In the case of startup jobs, we might be willing to optionally allow the user to interrupt the start of the app if a crash happens during their execution.

For instance, given the context of a "migration", if this doesn't work, there's no value in letting the app start, isn't there?

Reply:
Fair - we might want to have some "Options" here as well that the user can pass in to configure the behavior.
Kind of what ASP.NET does. So that you can easier pass them in here or via the `Configure` method.

But all in all I do see that I shouldn't directly commit on `v4` branch but make a feature branch, otherwise here is too much noise